test/e2e/multikueue/e2e_test.go

```diff
@@ -60,36 +83,174 @@ var _ = ginkgo.Describe("MultiKueue", func() {
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("Creating the job", func() {
+				gomega.Expect(k8sManagerClient.Create(ctx, job)).Should(gomega.Succeed())
+			})
 
-			// create an wl with the remote client
-			wl := utiltesting.MakeWorkload("wl", (*ns).Name).Obj()
-			gomega.Expect(remClient.Create(ctx, wl)).To(gomega.Succeed())
+			createdLeaderWorkload := &kueue.Workload{}
+			wlLookupKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(job.Name), Namespace: managerNs.Name}
 
-			createdWl := &kueue.Workload{}
-			gomega.Expect((*c).Get(ctx, client.ObjectKeyFromObject(wl), createdWl)).To(gomega.Succeed())
-			gomega.Expect(createdWl.Spec).To(gomega.BeComparableTo(wl.Spec))
+			// the execution should be given to the worker
+			ginkgo.By("Waiting to be admitted in worker1", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sManagerClient.Get(ctx, wlLookupKey, createdLeaderWorkload)).To(gomega.Succeed())
+					g.Expect(workload.FindAdmissionCheck(createdLeaderWorkload.Status.AdmissionChecks, multiKueueAc.Name)).To(gomega.BeComparableTo(&kueue.AdmissionCheckState{
+						Name:    multiKueueAc.Name,
+						State:   kueue.CheckStatePending,
+						Message: `The workload got reservation on "worker1"`,
+					}, cmpopts.IgnoreFields(kueue.AdmissionCheckState{}, "LastTransitionTime")))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
 
-		},
-			ginkgo.Entry("worker1", &k8sWorker1Client, "worker1.kubeconfig", &worker1Ns),
-			ginkgo.Entry("worker2", &k8sWorker2Client, "worker2.kubeconfig", &worker2Ns),
-		)
-	})
+			ginkgo.By("Waiting for the job to finish ", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sManagerClient.Get(ctx, wlLookupKey, createdLeaderWorkload)).To(gomega.Succeed())
 
+					g.Expect(apimeta.FindStatusCondition(createdLeaderWorkload.Status.Conditions, kueue.WorkloadFinished)).To(gomega.BeComparableTo(&metav1.Condition{
+						Type:    kueue.WorkloadFinished,
+						Status:  metav1.ConditionTrue,
+						Reason:  "JobFinished",
+						Message: `From remote "worker1": Job finished successfully`,
+					}, cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime")))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("The job remains completed", func() {
+				gomega.Consistently(func(g gomega.Gomega) {
+					createdJob := &batchv1.Job{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(job), createdJob)).To(gomega.Succeed())
+					g.Expect(ptr.Deref(createdJob.Spec.Suspend, false)).To(gomega.BeTrue())
+					g.Expect(createdJob.Status.Conditions).To(gomega.ContainElement(gomega.BeComparableTo(
+						batchv1.JobCondition{
+							Type:   batchv1.JobComplete,
+							Status: corev1.ConditionTrue,
+						},
+						cmpopts.IgnoreFields(batchv1.JobCondition{}, "LastTransitionTime", "LastProbeTime"))))
+				}, util.ConsistentDuration, util.Timeout).Should(gomega.Succeed())
+			})
```

Comment:
I would just use a one time check, after checking that worker objects were cleaned up.